lib/internal/test_runner/runner.js

```diff
@@ -432,6 +442,11 @@ function watchFiles(testFiles, root, inspectPort, testNamePatterns) {
       triggerUncaughtException(error, true /* fromPromise */);
     }));
   });
+  signal?.addEventListener('abort', () => {
```

Comment:
need to remove this listener if abort never happens but the watcher is already cleared or is this always before the process dies and "hanging" this if we never abort is fine?

Reply:
it is ok - if you don't pass a signal the watcher will never clear/close

Reply:
indeed - in watch mode `run` **does not finish** besides this new flow